extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelNativeImageProcessor.java

```diff
@@ -0,0 +1,298 @@
+import org.apache.camel.quarkus.core.deployment.spi.CamelServiceBuildItem;
+import org.apache.camel.quarkus.core.deployment.spi.CamelServicePatternBuildItem;
+import org.apache.camel.quarkus.core.deployment.util.CamelSupport;
+import org.apache.camel.quarkus.core.deployment.util.PathFilter;
+import org.apache.camel.spi.DataFormat;
+import org.apache.camel.spi.ExchangeFormatter;
+import org.apache.camel.spi.PropertiesComponent;
+import org.apache.camel.spi.ScheduledPollConsumerScheduler;
+import org.apache.camel.spi.StreamCachingStrategy;
+import org.apache.camel.support.CamelContextHelper;
+import org.jboss.jandex.AnnotationTarget.Kind;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.commons.lang3.ClassUtils.getPackageName;
+
+public class CamelNativeImageProcessor {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CamelNativeImageProcessor.class);
+
+    private static final List<Class<?>> CAMEL_REFLECTIVE_CLASSES = Arrays.asList(
+            Endpoint.class,
+            Consumer.class,
+            Producer.class,
+            TypeConverter.class,
+            ExchangeFormatter.class,
+            ScheduledPollConsumerScheduler.class,
+            Component.class,
+            CamelContext.class,
+            StreamCachingStrategy.class,
+            StreamCachingStrategy.SpoolUsedHeapMemoryLimit.class,
+            PropertiesComponent.class,
+            DataFormat.class);
+
+    @BuildStep
+    void reflectiveItems(
+            CombinedIndexBuildItem combinedIndex,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
+            BuildProducer<ReflectiveMethodBuildItem> reflectiveMethod) {
+
+        final IndexView view = combinedIndex.getIndex();
+
+        CAMEL_REFLECTIVE_CLASSES.stream()
+                .map(Class::getName)
+                .map(DotName::createSimple)
+                .map(view::getAllKnownImplementors)
+                .flatMap(Collection::stream)
+                .filter(CamelSupport::isPublic)
+                .forEach(v -> reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, v.name().toString())));
+
+        DotName converter = DotName.createSimple(Converter.class.getName());
+        List<ClassInfo> converterClasses = view.getAnnotations(converter)
+                .stream()
+                .filter(ai -> ai.target().kind() == Kind.CLASS)
+                .filter(ai -> {
+                    AnnotationValue av = ai.value("loader");
+                    boolean isLoader = av != null && av.asBoolean();
+                    // filter out camel-base converters which are automatically inlined in the
+                    // CoreStaticTypeConverterLoader
+                    // need to revisit with Camel 3.0.0-M3 which should improve this area
```

Comment:
Yeah all converters are now non-reflective, so we should look at this in another ticket